she/repair_pr.py

```
@@ -0,0 +1,206 @@
+    required_tests: tuple[str, ...]
+    rollback_sha: str
+    promotion_ready: bool = False
+    live: bool = False
+    mutates_source: bool = False
+    constraints: tuple[str, ...] = ()
+    metadata: dict[str, Any] = field(default_factory=dict)
+
+    def __post_init__(self) -> None:
+        needed = set(self.required_tests)
+        if not DUAL_GATES.issubset(needed):
+            raise RepairPRError("dual gates repo-gate + termux-smoke must be required tests")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)